### DISCUSSION TITLE
How to specify the simulation density in OMEdit?

### DISCUSSION THREAD

**Discussion by computational-physcis-2011301020083** [OpenModelica/OpenModelica#10045] [Q&A] [answered]
I try to simulate a quite complex mechanics system, and it failed and showed:
**because density of 0.022 remains under threshold of  0.200.
Using sparse solver for linear system 15,
because density of 0.22 remians under threshold of 0.200.
The maximum density for using sparse solvers can be specified 
using the runtime flag '<-lssMasDensity>=value'.**  

So how to adjust this density in OMEdit interface? or how to solve this?
Thanks!

**Comment by mahge** [comment, marked as answer]
Once you have opened your model,
- Go to simulation setup ( then [s] icon on the toolbar), 
- Then go to the _Simulation Flags_ tab
- Scroll to the bottom to find the `Additional Simulation Flags (Optional):` entry.
- Add your simulation flag, i.e., `-lssMasDensity=<value>`

You can ask OMEdit to save these flags in the models itself by ticking the checkbox `Save simulation flags inside model i.e., __OpenModelica_simulationFalgs annotation`. 

If you open your model afterwards you can see another way of specifying these options using annotations.

@AnHeuermann is there a better way of specifying these simFalgs globally  similar to the one for Translation flags in _Tools -> Options -> Simulation -> Additional Translation Flags:_

**Reply by AnHeuermann** [reply]
I don't think there is a global way to specify simulation flags for every model. So there is no equivalent to `Tools -> Options -> Simulation -> Additional Translation Flags` for simulation flags.

**Comment by casella** [comment]
Anyway, the notification you report just says that a sparse solver has been used instead of a dense one, because the system is not dense. Unless there are some bugs in OpenModelica, this has nothing to do with your simulation failing.
